test/e2e/declare_consumer.test.ts

```diff
@@ -178,9 +179,12 @@ describe("declare consumer", () => {
 
   it(`consume a lot of messages`, async () => {
     const receivedMessages: Buffer[] = []
-    await client.declareConsumer({ stream: streamName, offset: Offset.next() }, (message: Message) => {
-      receivedMessages.push(message.content)
-    })
+    await client.declareConsumer(
+      { stream: streamName, offset: Offset.next(), creditPolicy: creditsOnChunkReceived(1, 1) },
```

Comment:
Why do we have to set a non-default `creditPolicy` here?

Reply:
TLDR: 
The default values for credits are now set to have two credits on startup and one credit on renewal.
Given how the node event loop handles async calls, and how the test is written, either we need to use exacly one credit for this test, we change the checked condition, or we change the default values.

Long version:

First of all, note that in the checked condition for this test, the order in which messages are handled by the callback matters.

In the previous implementation (https://github.com/coders51/rabbitmq-stream-js-client/blob/16d26ff6ccc03485e00388fbdab3c506caea077d/src/client.ts) on startup 10 credits are assigned to the client. The message handling is implemented using a `.map` call, which does not allow for async behavior; for the node event loop this is basically an atomic synchronous call.

In this PR this has been updated with a `for{}` cycle to allow the async call to `onChunkProgress`.

The presence of this async call is enough for the event loop to allow the handling of received chunks to be interleaved. Note that only the handling on the client side is affected, the delivery order from RabbitMQ remains consistent with what was received from the publisher.

Let's start from the code in the main branch (so the new added logic is not involved) and substitute the call
```
response.messages.map((x) => consumer.handle(x))
```

to something like

```
for (const m of response.messages) {
    consumer.handle(m)
    await new Promise((res) => {
        res(true)
    })
}
```

which at first glance should behave exactly the same; the test will actually fail.